### PR TITLE
[BEAM-4321] Enforce ErrorProne analysis in join-library extensions

### PR DESCRIPTION
**Pull request by iemejia**
Randomly chosen reviewer
R: @lukecwik

### CODE REVIEW

**Comment by timrobertson100** [comment]
LGTM

File: sdks/java/extensions/join-library/build.gradle [hunk 1]
```diff
@@ -17,14 +17,16 @@
...
 description = "Apache Beam :: SDKs :: Java :: Extensions :: Join library"
 
 dependencies {
   compile library.java.guava
+  compileOnly library.java.findbugs_annotations
   shadow project(path: ":beam-sdks-java-core", configuration: "shadow")
   testCompile project(path: ":beam-runners-direct-java", configuration: "shadow")
   testCompile library.java.hamcrest_core
   testCompile library.java.junit
+  testCompileOnly library.java.findbugs_annotations
```
Comment (lukecwik) [inline sdks/java/extensions/join-library/build.gradle]: Why did you need to add the annotations to the compileOnly and testCompileOnly scopes when none are used below?

Comment (timrobertson100) [reply sdks/java/extensions/join-library/build.gradle]: Without them, you get a warning which (now) fails the build 

```
warning: Cannot find annotation method 'value()' in type 'DefaultAnnotation': class file for edu.umd.cs.findbugs.annotations.DefaultAnnotation not found
```

It comes from `package-info.class` stuff in upstream shaded jars.

There could be other/better ways of fixing this, but I couldn't find one so I started adding that in the IOs I fixed. I see others are following that approach.

Comment (iemejia) [reply sdks/java/extensions/join-library/build.gradle]: Yes is exactly as Tim says, it is the pattern we have followed so far in the other IOs because fo complains in package-info and extra warnings.

Comment (lukecwik) [reply sdks/java/extensions/join-library/build.gradle]: This looks fine, but overall it seems as though we could just make this a default dependency in build_rules.gradle whenever errorprone is enabled.

```
if (configuration.enableErrorProne) {
  dependencies {
    compileOnly  ...
    testCompileOnly ...
  }
}
```

Comment (iemejia) [reply sdks/java/extensions/join-library/build.gradle]: Ok, I or Tim will provide a PR to do this and remove the deps in all the modules we have already done. Thanks Luke.
